electron/gog/user.ts

```diff
@@ -0,0 +1,136 @@
+import axios from 'axios'
+import Store from 'electron-store'
+import { BrowserWindow } from 'electron'
+import { logError, logInfo, LogPrefix } from '../logger/logger'
+import { gogLoginUrl } from '../constants'
+import { GOGLoginData } from '../types'
+
+const configStore = new Store({
+  cwd: 'gog_store'
+})
+
+const gogEmbedRegExp = new RegExp('https://embed.gog.com/on_login_success?')
+const gogRefreshTokenUrl =
+  'https://auth.gog.com/token?client_id=46899977096215655&client_secret=9d85c43b1482497dbbce61f6e4aa173a433796eeae2ca8c5f6129f2dc4de46d9&grant_type=refresh_token'
+
+export class GOGUser {
+  // This is executed below in edge error cases most likely won't run ever
+  public static async handleGOGLogin() {
+    const popupWindow = new BrowserWindow({
+      width: 450,
+      height: 700,
+      title: 'GOG'
+    })
+    popupWindow.loadURL(gogLoginUrl)
+    popupWindow.once('ready-to-show', popupWindow.show)
+
+    popupWindow.webContents.on('did-finish-load', () => {
+      const pageUrl = popupWindow.webContents.getURL()
+      if (pageUrl.match(gogEmbedRegExp)) {
+        const parsedURL = new URL(pageUrl)
+        const code = parsedURL.searchParams.get('code')
+        this.login(code)
+        popupWindow.close()
+      }
+    })
+  }
+
+  static async login(code: string) {
+    logInfo('Logging using GOG credentials', LogPrefix.Gog)
+
+    // Gets token from GOG basaed on authorization code
+    const response = await axios
+      .get(
+        `https://auth.gog.com/token?client_id=46899977096215655&client_secret=9d85c43b1482497dbbce61f6e4aa173a433796eeae2ca8c5f6129f2dc4de46d9&grant_type=authorization_code&redirect_uri=https%3A%2F%2Fembed.gog.com%2Fon_login_success%3Forigin%3Dclient&code=${code}`
```

Comment:
Perhaps put this URL into a variable to make it more readable. 
I think it is too much too put as a param to a function. But is up to you.